File: SAMPLES/PYTHON/test_classes.py
# Test TODO 18 : Classes (class) et objets

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def sum(self):
        return self.x + self.y

p = Point(3, 4)
print(p.get_x())
print(p.get_y())
print(p.sum())

class Animal:
    def __init__(self, name):
        self.name = name
        self.sound = 0

    def speak(self):
        return self.sound

class Dog(Animal):
    def __init__(self, name):
        self.name = name
        self.sound = 42

d = Dog("Rex")
print(d.speak())

File: SAMPLES/KOTLIN/test_classes.kt
// test_classes.kt - Test classes et data classes (TODO 16)

class Point(val x: Int, val y: Int)

data class Person(val name: String, val age: Int)

val p = Point(10, 20)
println(p.x)
println(p.y)

val alice = Person("Alice", 30)
println(alice)
println(alice.name)
println(alice.age)

val bob = Person("Bob", 25)
println(bob)
